lib/src/mount.rs

```diff
@@ -146,6 +146,23 @@ pub(crate) fn is_same_as_host(path: &Utf8Path) -> Result<bool> {
     Ok(devstat.f_fsid == hostdevstat.f_fsid)
 }
 
+/// Open the named path as a new mount attached to an fd.
+#[context("Opening mount tree from pid")]
+pub(crate) fn open_tree(path: &Utf8Path, recursive: bool) -> Result<OwnedFd> {
+    // Open the target mount path as a file descriptor.
+    let recursive = if recursive {
+        OpenTreeFlags::AT_RECURSIVE
+    } else {
+        OpenTreeFlags::empty()
+    };
+    rustix::mount::open_tree(
+        rustix::fs::CWD,
+        path.as_std_path(),
+        OpenTreeFlags::OPEN_TREE_CLOEXEC | OpenTreeFlags::OPEN_TREE_CLONE | recursive,
```

Comment:
Also per https://github.com/brauner/man-pages-md/pull/15 we can't actually use `OPEN_TREE_CLONE` for the linting case because the default for docker/podman is not to be in a user namespace with `CAP_SYS_ADMIN`.

(Eventually of course it'd make huge amounts of sense to try to drive a userns to be enabled by default for builds and runs but it has a lot of potential fallout)

Reply:
I think we could do this if we created our own user and mount namespace.  That works for `bwrap`, even as a non-root user, and even running inside of a container itself.  It's starting to get a bit "heavy", but as a general feature to have inside of bootc, it would be quite useful.  I already imagine some uses for it in composefs-rs as well, for the "unified" example...

Of course, we already have the code in `mount.rs` for entering another mount namespace and sending a mount fd out of it back to ourselves... this would probably end up looking quite a lot like that, but instead of entering a new namespace, we create our own...